web/src/echo.ts

```diff
@@ -171,7 +171,6 @@ function resend_message(
     function on_success(raw_data: unknown): void {
         const data = send_message_api_response_schema.parse(raw_data);
         const message_id = data.id;
-        message.locally_echoed = true;
```

Comment:
Yeah, this bug looks to have been introduced in d1f82c47fd08a5df55f93a669590450a07c8fea6, @prakhar1144 FYI. The hunk in this file is:
``` diff
diff --git a/web/src/echo.js b/web/src/echo.js
index ea2f9316b7..bd337cf288 100644
--- a/web/src/echo.js
+++ b/web/src/echo.js
@@ -84,15 +84,13 @@ function resend_message(message, $row, {on_send_message_success, send_message})
     message.queue_id = page_params.queue_id;
     message.resend = true;
 
-    const local_id = message.local_id;
-
     function on_success(data) {
         const message_id = data.id;
-        const locally_echoed = true;
+        message.locally_echoed = true;
 
         hide_retry_spinner($row);
 
-        on_send_message_success(local_id, message_id, locally_echoed);
+        on_send_message_success(message, data);
 
         // Resend succeeded, so mark as no longer failed
         failed_message_success(message_id);
```

Overall that function has some serious variable reuse problems that will be very apparent when we try to convert `transmit.js` to TypeScript; it's converting a `Message` into send-request by mutating it. @evykassirer FYI since you'v been cleaning up a lot of similar TS messes.